dateutil/test/test_rrule.py

```diff
@@ -4614,6 +4614,14 @@ def testReplaceIfNotSet(self):
         self.assertEqual(list(newrr),
                              [datetime(1997, 1, 6)])
 
+    def testRaisesCorrectErrorForUnsupportedProperty(self):
+        rule = "X-ABC-MMSUBJ:19970902T090000"
+        assertRaisesRegex(self, ValueError, "unsupported property", rrulestr, rule, dtstart=datetime(1997, 9, 2, 9, 0))
```

Comment:
I don't care about this too much, but I'd prefer if we used `pytest` here instead of the unittest-specific stuff. I also strongly prefer a context manager form. The `pytest` version of this is:

```python
with pytest.raises(ValueError, match="unsupported property"):
    rrulestr(rule, dtstart=datetime(1997, 9, 2, 9, 0))
```

That said, there are two other issues here:

1. We care less about the "unsupported property" message than that we've picked out the _correct_ unsupported property.
2. The `dtstart=datetime(1997, 9, 2, 9, 0)` portion of this is unnecessary, and including it might confuse the reader into thinking that it's an important part of the test.

So, I suggest refactoring this test like so:

```python
def testRaisesCorrectErrorForUnsupportedProperty(self):
    with pytest.raises(ValueError, match=re.escape("X-ABC-MMSUBJ")):
        rrulestr("X-ABC-MMSUBJ:19970902T090000")
```

Reply:
Got it, will change accordingly